esphome/components/micro_wake_word/__init__.py

```diff
@@ -319,42 +412,83 @@ async def to_code(config):
+        config[CONF_MODELS].append(vad_model)
 
-    manifest, data = _load_model_data(file)
+    for model_parameters in config[CONF_MODELS]:
+        model_config = model_parameters.get(CONF_MODEL)
+        data = []
+        if model_config[CONF_TYPE] == TYPE_GIT:
+            # compute path to model file
+            key = f"{model_config[CONF_URL]}@{model_config.get(CONF_REF)}"
+            base_dir = Path(CORE.data_dir) / DOMAIN
+            h = hashlib.new("sha256")
+            h.update(key.encode())
+            file: Path = base_dir / h.hexdigest()[:8] / model_config[CONF_FILE]
 
-    rhs = [HexInt(x) for x in data]
-    prog_arr = cg.progmem_array(config[CONF_RAW_DATA_ID], rhs)
-    cg.add(var.set_model_start(prog_arr))
+        elif model_config[CONF_TYPE] == TYPE_LOCAL:
+            file = Path(model_config[CONF_PATH])
 
-    probability_cutoff = config.get(
-        CONF_PROBABILITY_CUTOFF, manifest[KEY_MICRO][CONF_PROBABILITY_CUTOFF]
-    )
-    cg.add(var.set_probability_cutoff(probability_cutoff))
-    sliding_window_average_size = config.get(
-        CONF_SLIDING_WINDOW_AVERAGE_SIZE,
-        manifest[KEY_MICRO][CONF_SLIDING_WINDOW_AVERAGE_SIZE],
-    )
-    cg.add(var.set_sliding_window_average_size(sliding_window_average_size))
+        elif model_config[CONF_TYPE] == TYPE_HTTP:
+            file = _compute_local_file_path(model_config) / "manifest.json"
+
+        else:
+            raise ValueError("Unsupported config type: {model_config[CONF_TYPE]}")
+
+        manifest, data = _load_model_data(file)
+
+        rhs = [HexInt(x) for x in data]
+        prog_arr = cg.progmem_array(model_parameters[CONF_RAW_DATA_ID], rhs)
+
+        probability_cutoff = model_parameters.get(
+            CONF_PROBABILITY_CUTOFF, manifest[KEY_MICRO][CONF_PROBABILITY_CUTOFF]
+        )
+        sliding_window_size = model_parameters.get(
+            CONF_SLIDING_WINDOW_SIZE,
+            manifest[KEY_MICRO][CONF_SLIDING_WINDOW_SIZE],
+        )
 
-    cg.add(var.set_wake_word(manifest[KEY_WAKE_WORD]))
+        if manifest[KEY_WAKE_WORD] == "vad":
+            cg.add(
+                var.add_vad_model(
+                    prog_arr,
+                    probability_cutoff,
+                    sliding_window_size,
+                    manifest[KEY_MICRO][CONF_TENSOR_ARENA_SIZE],
+                )
+            )
+        else:
+            cg.add(
+                var.add_wake_word_model(
+                    prog_arr,
+                    probability_cutoff,
+                    sliding_window_size,
+                    manifest[KEY_WAKE_WORD],
+                    manifest[KEY_MICRO][CONF_TENSOR_ARENA_SIZE],
+                )
+            )
+
+        model_step_size = manifest[KEY_MICRO][CONF_FEATURE_STEP_SIZE]
+
+        if features_step_size is None:
+            features_step_size = model_step_size
+        elif features_step_size != model_step_size:
+            raise cv.Invalid("Cannot load models with differet features step sizes.")
```

Comment:
We cannot raise cv.Invalid errors inside `to_code`. This should be checked in the validation stage.
Also a typo =)

```suggestion
            raise cv.Invalid("Cannot load models with different features step sizes.")
```

Reply:
I've created a new function to validate this and just set ``FINAL_VALIDATE_SCHEMA = _feature_step_size_validate`` I think this is the right approach from looking at other components, but I'm not 100% sure.